spec/plsql/package_spec.rb

```diff
@@ -67,6 +66,43 @@
     end
   end
 
+  context "with a user with execute privilege who is not the package owner" do
+    before(:all) do
+      plsql.execute("grant execute on TEST_PACKAGE to #{DATABASE_USERS_AND_PASSWORDS[1][0]}")
+      @original_connection = plsql.connection
+      @conn = get_connection(1)
+    end
+
+    before(:each) do
+      # resetting connection clears cached package objects and schema name
+      plsql.connection = @conn
+    end
+
+    it "should not find existing package" do
+      expect(PLSQL::Package.find(plsql, :test_package)).to be_nil
+    end
+
+    context "who sets current_schema to match the package owner" do
+      before(:all) do
+        plsql.execute "ALTER SESSION set current_schema=#{DATABASE_USERS_AND_PASSWORDS[0][0]}"
+      end
+
+      it "should find existing package" do
+        expect(PLSQL::Package.find(plsql, :test_package)).not_to be_nil
+      end
+
+      it "should report an existing procedure as existing" do
+        expect(plsql.test_package.procedure_defined?(:test_procedure)).to be_truthy
+      end
+
+    end
+
+    after(:all) do
```

Comment:
I wonder if would be better to have `after(:all)` close to the relevant `before` block (it feels a bit easier to see what we need to tear down).